Lib/unittest/test/test_case.py

```diff
@@ -1973,5 +1976,234 @@ def test2(self):
+        # ----------------------
+
+        def run_case(case: Case, N):
+            """Given one of the test cases above, execute the test case for a
+            given `N` constant value. Check if the test has passed as
+            specified."""
+
+            # we are working our way towards _heuristic_diff(foo, bar)
+
+                    # --- Construct Differing Strings ---
+
+            # construct foo. Double line count because bar will have twice
+            # as many lines (lines of 'a' and lines of 'b')
+            foo = (
+                ('a' * N * (case.line_length_factor),)  # create line
+                * (N * case.num_lines_factor * 2)       # duplicate line
+            )
+
+            # construct bar
+            bar_a_line = ('a' * (N * case.line_length_factor))
+            bar_b_line = ('b' * (N * case.line_length_factor))
+            bar_a_lines = ((bar_a_line,) * (N * case.num_lines_factor))
+            if case.extent_differing != 1:
+                # diminish the amount of 'b' by case.extent_differing, and add
+                # additional 'a' at the end as padding
+                bar_b_lines = ((bar_b_line,)
+                               * int(N
+                                     * case.num_lines_factor
+                                     * case.extent_differing))
+                bar_a_padding = ((bar_a_line,)
+                                 * int(N
+                                       * case.num_lines_factor
+                                       * (1 - case.extent_differing)))
+                bar = (*bar_a_lines, *bar_b_lines, *bar_a_padding)
+            else:
+                bar_b_lines = ((bar_b_line,) * int(N * case.num_lines_factor))
+                bar = (*bar_a_lines, *bar_b_lines)
+
+                    # --- Perform Diff ---
+
+            # after all that, we have `foo` and `bar`; two string sequences
+            # with differences as specified by the Case parameters, scaled
+            # by a factor of `N`.
+            diff = unittest.case._heuristic_diff(foo, bar)
+
+                    # --- Make Assertions ---
+
+            # now, check that the `case.expect_unified_diff_at` condition was
+            # met
+            if (
+                N < case.expect_unified_diff_at
+                    and self.is_unified_diff(diff)
+            ):
+                self.fail('Switched to `unified_diff` prematurely. Expected '
+                          f'switch at {case.expect_unified_diff_at}, but '
+                          f'actually switched at {N} for the case {case}')
+            elif (
+                N > case.expect_unified_diff_at
+                and not self.is_unified_diff(diff)
+            ):
+                self.fail('Switch to `unified_diff` did not occur. Expected '
+                          'switch at {case.expect_unified_diff_at}, but no '
+                          f'switch occured when N == {N}')
+
+        for case in cases:
```

Comment:
You're basically implementing test case parameterization here within one test. I woefully miss things like https://github.com/abseil/abseil-py/blob/main/absl/testing/parameterized.py or pytest's parameterized in the stdlib...

anyways, not a big deal.  this will work.  but a future refactoring when we support some form of easy parameterization in the stdlib would generate ~100 different test methods out of this. :)